rollout.go

```diff
@@ -64,29 +62,32 @@ func (r *client) Stop() {
 
 func (r *client) poll(path string) {
 	defer func() { r.done <- true }()
-	defer log.Println("Rollout poller shutdown")
+	defer log.Info("Rollout poller shutdown")
 	for {
 		data, _, watch, err := r.zk.GetW(path)
 		if err != nil {
-			log.Println("Rollout: Failed to set watch", err)
+			log.Error("Rollout: Failed to set watch", err)
 			if r.errorHandler != nil {
 				r.errorHandler(err)
 			}
+
 			select {
-				case <- time.After(time.Second):
-				case <- r.stop:
-					return
-				}
+			case <-time.After(time.Second):
+			case <-r.stop:
+				return
+			}
 			continue
 		}
-		newMap := make(map[string]string)
-		err = json.Unmarshal([]byte(data), &newMap)
-		if err != nil {
-			log.Println("Rollout: Couldn't unmarshal zookeeper data", err)
+
+		if err := r.swapData(data); err != nil {
+			log.Error("Rollout: Couldn't unmarshal zookeeper data", err)
+			// re-get the watch so we know when/if the bad data changes
+			_, _, watch, err = r.zk.GetW(path)
+			if err != nil {
+				log.Fatal("could not re-establish a watch after unmarshalling error. Nothing to do but exit")
+			}
```

Comment:
Yup, good catch. You're right, the `GetW()` at the top of the loop should do the trick. I'll remove this block.

Reply:
Persons can/do go in there and edit the feature flag map by hand, I"m told, which is where these errors would likely come from. I'm glad to *not* put an empty map there since that'd break... well, all sorts of stuff.